src/Bicep.LangServer/Handlers/BicepDeploymentStartCommandHandler.cs

```diff
@@ -63,6 +78,36 @@ public override async Task<BicepDeploymentStartResponse> Handle(BicepDeploymentS
             var credential = new CredentialFromTokenAndTimeStamp(request.token, request.expiresOnTimestamp);
             var armClient = new ArmClient(credential, default, options);
 
+            string? bicepparamJsonOutput = null;
+
+            if(request.parametersFilePath.EndsWith(".bicepparam"))
+            {
+                //params file validation 
+                if(request.parametersFileUpdateOption != ParametersFileUpdateOption.None)
+                {
+                    return new BicepDeploymentStartResponse(false, "Can not create/overwrite/update parameter files when using a bicep parameters file", null);
+                }
+
+                if(request.updatedDeploymentParameters.Any())
+                {
+                    return new BicepDeploymentStartResponse(false, "Can not update parameters for bicep parameter file", null);
+                }
+
+                //params file compilation
+                var documentUri = DocumentUri.FromFileSystemPath(request.parametersFilePath);
+                var compilation = await new CompilationHelper(bicepCompiler, compilationManager).GetCompilation(documentUri);
+                
+                var diagnosticsByFile = compilation.GetAllDiagnosticsByBicepFile()
```

Comment:
I'm wondering if we could just do `compilation.GetEntrypointSemanticModel().HasErrors()` to check for errors. If not, could you extract the functionality to check for errors across all semantic models into a reusable extension method on the `Compilation` class?